### PR TITLE
🚮 delete temp file when test/deprecated.jl is done

### PR DESCRIPTION
**Pull request by tkelman**

### CODE REVIEW

**Comment by ararslan** [comment]
+1 for the change and for the commit message.

**Comment by tkelman** [comment]
There are several in `test/io.jl` that are also littering, they delete the file if it exists immediately after creating it but don't delete it before overwriting the variable that contained the filename.

File: test/deprecated.jl [hunk 1]
```diff
@@ -1,8 +1,9 @@
 using DataFrames, FileIO, RDatasets, Compat
-fn = tempname()
-open(fn, "w") do f
-    redirect_stderr(f) do
-        dataset("datasets", "iris")
+mktemp() do fn
```
Comment (ararslan) [inline test/deprecated.jl]: Looks like the syntax or something is off, maybe? This line is causing failures on all versions and platforms.

**Comment by ararslan** [comment]
> There are several in test/io.jl that are also littering

If you wanted to tackle those as well in this PR, that would be much appreciated :wink:

File: test/deprecated.jl (hunk 1 above)
Comment (tkelman) [inline test/deprecated.jl]: oops, sorry for wasting your ci time. danger of web-ui PR's

Comment (andreasnoack) [inline test/deprecated.jl]: Didn't know about `mktemp`. Anyway, I had to close the file before I could call `readstring` so `mktemp` might not work in this case.

**Pull request merged**